packages/prisma/extensions/exclude-locked-users.ts

```diff
@@ -35,8 +43,10 @@ async function excludeLockedUsers(
   query: <T>(args: T) => Promise<unknown>
 ) {
   args.where = args.where || {};
+  const whereString = safeJSONStringify(args.where);
+  const shouldIncludeLocked = whereString.includes('"locked":false');
   // Unless explicitly specified, we exclude locked users
-  if (args.where.locked === undefined) {
+  if (!shouldIncludeLocked) {
```

Comment:
This only checked for "locked" at top level in the where property. The new way includes nested AND and OR conditions.